types/promise-hash/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for promise-hash 1.3
+// Project: https://github.com/mtimofiiv/promise-hash
+// Definitions by: Michael Shafir <https://github.com/mshafir>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = hash;
+
+declare global {
+    interface PromiseConstructor {
+        hash<T>(promiseHash: { [P in keyof T]: PromiseLike<T[P]> | T[P] }):
+            Promise<{ [P in keyof T]: T[P] }>;
+    }
+}
+
+declare function hash<T>(
+    promiseHash: { [P in keyof T]: PromiseLike<T[P]> | T[P]}):
+    Promise<{ [P in keyof T]: T[P] }>;
```

Comment:
```suggestion
    Promise<T>;
```

Reply:
Making this change makes the tests fail, though it's not entirely clear to me why.
`ERROR: 12:1  expect  TypeScript@next expected type to be:
  Promise<{ a: number; b: string; c: { d: boolean; e: string; }; }>
got:
  Promise<{ a: number; b: string; c: { d: any; e: any; }; }>`